test/test_sparse_csr.py

```diff
@@ -930,6 +931,18 @@ def test_csr_is_contiguous(self):
         with self.assertRaisesRegex(RuntimeError, "Sparse CSR tensors do not have is_contiguous"):
             a.is_contiguous()
 
+    @onlyCPU
+    @largeTensorTest("20GB", "cpu")
+    def test_csr_nnz(self):
+        # Tests the limits of the number of specified elements in CSR tensors, see gh-102520.
+        for nnz in [0, 2**31]:
+            rows, cols = 1, max(nnz, 1)
+            crow_indices = torch.tensor([0, nnz], dtype=torch.int64)
+            col_indices = torch.arange(nnz, dtype=torch.int64)
+            values = torch.ones(nnz, dtype=torch.int8)
+            a = torch.sparse_csr_tensor(crow_indices, col_indices, values, (rows, cols))
+            self.assertEqual(a._nnz(), nnz)
```

Comment:
While grepping for `int nnz` I found the following lines:
https://github.com/pytorch/pytorch/blob/0f1621df1a0a73956c7ce4e2f72f069e610e0137/aten/src/ATen/native/sparse/cuda/SparseBlasImpl.cpp#L1123-L1125
Would you be interested in sending a fix in this stack? Seems like changing the line to
`max(nnz, at::native::cuda_int_cast(C_crow_indices.narrow(-1, m, 1).item<int64_t>(), "nnzC")` could solve the issue...

Reply:
Ah, sorry, the fix is wrong :) But the issue is still there

Reply:
The code that you referenced, is working in 32-bit integer mode only. So, this (supporting 64-bit indices in add_out_sparse_csr) is really not an issue (it does not seem to be reproducible) but a lack of feature support that should be tackled separately.

Reply:
Interesting, the code takes in an out tensor, converts it to int32 in-place, then just accesses nnz without any checks. These nnz checks are there for the inputs, however, so this code can still fail, so it seems... But all right, this could be fixed later on....